packages/ember-htmlbars/tests/helpers/if_unless_test.js

```diff
@@ -1,260 +0,0 @@
-    template: compile('{{#if view.show}}{{this}}{{/if}}'),
-    show: true
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'abc', 'should start property - precond');
-
-  run(function() {
-    view.set('show', false);
-  });
-
-  equal(view.$().text(), '');
-});
-
-QUnit.test('properties within an if statement should not fail on re-render', function() {
-  view = EmberView.create({
-    template: compile('{{#if view.value}}{{view.value}}{{/if}}'),
-    value: null
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), '');
-
-  run(function() {
-    view.set('value', 'test');
-  });
-
-  equal(view.$().text(), 'test');
-
-  run(function() {
-    view.set('value', null);
-  });
-
-  equal(view.$().text(), '');
-});
-
-QUnit.test('the {{this}} helper should not fail on removal', function() {
-  view = EmberView.create({
-    context: 'abc',
-    template: compile('{{#if view.show}}{{this}}{{/if}}'),
-    show: true
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'abc', 'should start property - precond');
-
-  run(function() {
-    view.set('show', false);
-  });
-
-  equal(view.$().text(), '');
-});
-
-QUnit.test('edge case: child conditional should not render children if parent conditional becomes false', function() {
-  var childCreated = false;
-  var child = null;
-
-  view = EmberView.create({
-    cond1: true,
-    cond2: false,
-    viewClass: EmberView.extend({
-      init() {
-        this._super(...arguments);
-        childCreated = true;
-        child = this;
-      }
-    }),
-    template: compile('{{#if view.cond1}}{{#if view.cond2}}{{#view view.viewClass}}test{{/view}}{{/if}}{{/if}}')
-  });
-
-  runAppend(view);
-
-  ok(!childCreated, 'precondition');
-
-  run(function() {
-    // The order of these sets is important for the test
-    view.set('cond2', true);
-    view.set('cond1', false);
-  });
-
-  // TODO: Priority Queue, for now ensure correct result.
-  ok(!childCreated, 'child should not be created');
-  //ok(child.isDestroyed, 'child should be gone');
-  equal(view.$().text(), '');
-});
-
-QUnit.test('edge case: rerender appearance of inner virtual view', function() {
-  view = EmberView.create({
-    tagName: '',
-    cond2: false,
-    template: compile('{{#if view.cond2}}test{{/if}}')
-  });
-
-  runAppend(view);
-  equal(jQuery('#qunit-fixture').text(), '');
-
-  run(function() {
-    view.set('cond2', true);
-  });
-
-  equal(jQuery('#qunit-fixture').text(), 'test');
-});
-
-QUnit.test('`if` helper with inline form: can use truthy param as binding', function() {
-  view = EmberView.create({
-    truthy: 'ok',
-    conditional: true,
-    template: compile('{{if view.conditional view.truthy}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'ok');
-
-  run(function() {
-    view.set('truthy', 'yes');
-  });
-
-  equal(view.$().text(), 'yes');
-});
-
-QUnit.test('using `if` with an `{{each}}` destroys components when transitioning to and from inverse (GH #12267)', function() {
```

Comment:
ported to `@test using `if` with an `{{each}}` destroys components when transitioning to and from inverse (GH #12267)`